File: my_pubsub/my_pubsub/hello_world.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node

class HelloWorld(Node):
    def __init__(self):
        self.counter=0

        super().__init__('hello_world_node')
        self.get_logger().info('Hello World of ROS2!')
        self.create_timer(1,self.timer_callback)

    def timer_callback(self):
        self.get_logger().info("Hello World! "+str(self.counter))
        self.counter+=1
    

def main(args=None):
    rclpy.init(args=args)
    node=HelloWorld()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__=='__main__':
    main()

File: cpp_pubsub/src/hello_world.cpp
#include "rclcpp/rclcpp.hpp"


class MyNode : public rclcpp::Node
{
public:
    MyNode() : Node("hello_world")
    {
        RCLCPP_INFO(this->get_logger(), "Hello Cpp Node");

        timer_ = this->create_wall_timer(std::chrono::seconds(1),
                                         std::bind(&MyNode::timerCallback, this));
    }

private:
    void timerCallback()
    {
        counter_++;
        RCLCPP_INFO(this->get_logger(), "Hello World! Im frm timerCB : %d", counter_);
    }

    rclcpp::TimerBase::SharedPtr timer_;
    int counter_=0;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<MyNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}